### ISSUE TITLE
SimpleDiscoveryClientAutoConfiguration doesn't use SimpleDiscoveryProperties bean, but direct method call

### ISSUE THREAD

**Issue by lightoze** [spring-cloud/spring-cloud-commons#687]
When I attempt to use service discovery from `BeanPostProcessor`, `SimpleDiscoveryClient` returns empty instance list. Example code would be:
```
applicationContext.getBean(CompositeDiscoveryClient.class).getInstances("oidc-server")
```

While debugging I can see that `SimpleDiscoveryClient#getInstances` is called before `SimpleDiscoveryProperties#init`, and during that first call properties are not yet initialized and instance list is empty.

Root cause seems to be `SimpleDiscoveryClientAutoConfiguration` and `proxyBeanMethods = false`. Should be trivial to fix with explicit method argument injection.

**Comment by spencergibb** [comment]
`SimpleDiscoveryProperties#init` is a bean lifecycle method that is only called after the bean is properly constructed. Using it in a `BeanPostProcessor` is simply too early.

**Comment by lightoze** [comment]
I'm using `postProcessAfterInitialization` which is called after lifecycle methods. The problem is `SimpleDiscoveryClientAutoConfiguration` directly calling own `@Bean` method while proxying is disabled.

**Comment by spencergibb** [comment]
right, looks like 18a4227c4ef1bdf86c3103e64bce75847a53073e broke this.

**Comment by spencergibb** [comment]
Actually closed by 0f78c5b4adf30d8f3f317e3f77c464572ee08628 not 69a8cec

**Comment by lightoze** [comment]
@spencergibb thanks!
